doc/specs/#3061 - Marketing Version.md

```diff
@@ -0,0 +1,124 @@
+---
+author: Kaleb Luedtke (Trenly) trenlymc@gmail.com
+created on: 2023-03-10
+last updated: 2023-03-10
+issue id: 3061
+---
+
+# Marketing Version
+
+"For [#3061](https://github.com/microsoft/winget-cli/issues/3061)"
+
+## Abstract
+
+There are times when the version which is marketed to consumers on publishers' websites differs from what is written to the Apps and Features Entries data. Additionally, there are times when the same version is marketed for several different versions of the same package. The Windows Package Manager should be able to show an arbitrary version as determined by the publisher.
+
+## Inspiration
+
+The current design using `PackageVersion` for the displayed version and how it differs from `DisplayVersion` can be confusing.
+
+## Solution Design
+
+The Windows Package Manager manifest schema should have a `MarketingVersion` key added at the root level. For clarity, it should be placed immediately following the `PackageVersion` key.
+
+When the Windows Package Manager needs to display the version of a package, the value from the `MarketingVersion` key should be used. If the key is not present or has no value, the value of `PackageVersion` should be displayed instead, with the exception of `winget show <query> --versions` and `winget search <query> --versions` which should show the values as key pairs of `PackageVersion (MarketingVersion)`.
+
+When the `--version` parameter is used, matches with `PackageVersion` should be considered first and matches with `MarketingVersion` should be considered second. If there are multiple package versions of the same identifier matched using `MarketingVersion`, the highest `PackageVersion` shall be selected.
+
+When `MarketingVersion` is present, `PackageVersion` should be validated to be a semantic version.
+
+For the purposes of version comparison, mapping installed packages to versions, and all other behaviors, the current implementation using `PackageVersion` should be retained.
+
+## UI/UX Design
+
+Example of a manifest with a `MarketingVersion` specified:
+```yaml
+PackageIdentifier: Example.Package
+PackageVersion: 1.2.5.1
+MarketingVersion: 1.2.5
+Installers:
+  InstallerUrl: https://example.com
+  InstallerSha256: <SHA>
+  InstallerType: exe
+  AppsAndFeaturesEntries:
+    - DisplayVersion: 1.2.5-abcde
+...
+```
+
+### `winget list`
+```
+Name                Id               Version   Available
+---------------------------------------------------------- 
+Example Package     Example.Package  1.2.4     1.2.5
```

Comment:
Just to confirm, the installed Version here is from arp entry display version and Available version is from marketing version? Would that cause confusion if Available version is way different from installed version?

Reply:
Or were you thinking the installed Version would be mapped to marketing version as well?

Reply:
This is assuming all installed versions are listed in winget repos, which is not the case most of the time. Currently installed to available mapping only happens when the manifest contains DisplayName in Apps and Features entries. If we want to achieve showing marketing version for installed, we need to map installed to available for all cases. And likely, we'll get `< SomeMarketVersion` as a result most of the time.

Reply:
Hmmm. That's true, but its also not different than what happens currently when there is a DisplayVersion specified and the installed version doesn’t match an available version

Reply:
Yep. We can only do relative version `< SomeVersion` if the mapping is performed. It's just the question that is showing `< SomeMarketVersion` is better or  `< SomePackageVersion` is better.

I was thinking to make the logic easier. We only do mapping when ARP DisplayVersion is specified in the manifest, most likely, packages that need marketing version are those packages needing mapping because they write non very standard versions in registry.

Reply:
That makes sense, and it would be a rarity to use MarketingVersion without DisplayVersion